src/main/java/com/icthh/xm/uaa/web/rest/AccountResource.java

```diff
@@ -271,6 +277,33 @@ public String isAuthenticated(HttpServletRequest request) {
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
+    /**
+     * POST /account/reset_password/init : Reset password and send reset key via customizable transport/flow.
+     *
+     * @param request password reset request
+     * @return the ResponseEntity with status 200 (OK) if the password was reset and reset flow (if configured) is executed
+     */
+    @PostMapping(path = "/account/reset_password/init", produces = MediaType.APPLICATION_JSON_UTF8_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Timed
+    @PreAuthorize("hasPermission({'login': #request.login}, 'ACCOUNT.PASSWORD.RESET')")
```

Comment:
hmm, seems like this permission will interfere with `hasPermission({'mail': #mail}, 'ACCOUNT.PASSWORD.RESET')` 

@ishkurko @antonysychov how are you plan to handle it in SPEL?

Reply:
as far as i understand, there should't be any issues with that
All previously created rules should work 
for instance 
    - privilegeKey: "ACCOUNT.PASSWORD.RESET"
      resourceCondition: "#mail != 'a@b.c"
will keep working 
and new ones can be added if one want to use login variable 
for instance 
    - privilegeKey: "ACCOUNT.PASSWORD.RESET"
      resourceCondition: "#mail != 'a@b.c && #login != '380501111111'"

if I get you wrong please clarify question a bit

Reply:
but you have two different APIs with different contexts... 
maybe it worth separating them by different permission keys?
@ishkurko @antonysychov what do you think?

Reply:
yes, endpoints are different but they relate to the same functionality in fact, that's why I decided to use the same permission key. It's discussable of course. @ishkurko what do think ?

Reply:
Right now, in xm used concept with one permission for the one API. This is why I am propose to separate. And discuss this concept in another thread as general xm architecture vision if needed.

Reply:
okay, really, lets separate keys in order not to make things complicated

ACCOUNT.PASSWORD.RESET_BY_TYPE - is this permission key for new endpoint is acceptable for everyone ? @ishkurko @amedvedchuk